### PR TITLE
fix: escape escape codes in the serializer

### LINKED ISSUE #21: Serializing paths on windows doesn't escape backslash
I'm using the serialize to store a path into a toml file. However when trying to parse that file back it fails. Looks like the serialize doesn't escape the backslash. For examples here is a generated toml. I tried both serialize with a wrapping struct and serializeKeyValue with the same results. The exe_path is stored in  a []u8.

```
exe_path = "D:\Projects\output\Runner.exe"
```

### PR DESCRIPTION
**Pull request by einetuer**
fixes #21

I've also noticed another issue which I fixed alongside this. Previously, when given a string like `hello\n` we would happily write this to the output stream:

```
key = "hello
"
```

which is of course very wrong.

### CODE REVIEW

**Comment by 0x5a4** [comment]
hmmm, how should we handle non ascii chars? should we just write them as is (the current behaviour)? or maybe convert them to the appropriate toml escape code `\uXXXX`

File: src/serializer.zig [hunk 1]
```diff
@@ -620,9 +634,18 @@ test "encode basic types" {
 
     // optionals
     try testWriteStream(@as(?u16, 42), "value", "value = 42\n");
+}
 
-    // strings
+test "encode strings" {
```
Comment (mattyhall) [inline src/serializer.zig:639]: Let's also test the Windows path specifically

**Review by mattyhall — CHANGES_REQUESTED** [review]
> hmmm, how should we handle non ascii chars?

The [docs say](https://github.com/toml-lang/toml/blob/main/toml.md#string):

> Any Unicode character may be used except those that must be escaped: quotation mark, backslash, and the control characters other than tab (U+0000 to U+0008, U+000A to U+001F, U+007F).

So I think we can leave them as is, apart from the control characters which we do need to escape.

File: src/serializer.zig [hunk 2]
```diff
@@ -373,7 +373,21 @@ pub fn WriteStream(
 
                         // This is a []const u8, or some similar Zig string.
                         if (ptr_info.child == u8 and std.unicode.utf8ValidateSlice(slice)) {
-                            return self.out_stream.print("\"{s}\"", .{value});
+                            try self.out_stream.writeByte('"');
+
+                            for (value) |byte| {
+                                switch (byte) {
+                                    '\\' => try self.out_stream.writeAll("\\"),
```
[the remark this answers is not in the document]
Comment (0x5a4) [reply src/serializer.zig]: i actually made this mistake as well, if we encounter a singular backslash we dont want to write two escaped backslashes, just one

[the remark this answers is not in the document]
Comment (0x5a4) [reply src/serializer.zig]: oh im stupid nevermind

**Review by mattyhall — APPROVED** [review]
Thanks!

**Comment by einetuer** [comment]
could you merge #17 as well?

**Pull request merged**